### DISCUSSION TITLE
Converting shapefile derived data in Tableau hyper file to valid geodataframe

### DISCUSSION THREAD

**Discussion by awa5114** [geopandas/geopandas#3469] [Q&A] [answered]
I'm attempting to read shapefile data stored in a Tableau HYPER extract. I can only retrieve this data by running an SQL query on the file as shown in their [docs](https://tableau.github.io/hyper-db/docs/guides/hyper_file/read). In my case this is what the resulting dataframe looks like:

<img width="700" alt="image" src="https://github.com/user-attachments/assets/63dd75a4-1530-4255-9067-0713e4f10130">

I'm pretty sure the geometry is not in WKB format, rather it comes in some sort of ESRI proprietary format. When I try to pass it to the `GeoDataFrame` constructor I get an error:

`gdf = gpd.GeoDataFrame(df, geometry="Geometry")`

> TypeError: Input must be valid geometry objects

Are there any alternative ways I can convert this to a GeoDataFrame, especially considering the fact I can apparently only get this data from the Hyper API in this bytestring format?

**Comment by martinfleis** [comment]
If `from_wkb` does not work then you will either need to get geometry in some other format or figure out yourself how to decode the bytes into something sensible.

**Comment by brendan-ward** [comment, marked as answer]
What data type does HYPER say that your `Geometry` column is?  I'm not finding a specific geometry type in their docs, but I am finding [Geography](https://tableau.github.io/hyper-db/lang_docs/py/tableauhyperapi.html#tableauhyperapi.TypeTag.GEOGRAPHY), so my first guess is that your `Geometry` column here contains bytes representing encoded an geography data type, not a 2d *geometry* type like you'd use for WKB/WKT as input to something like GeoPandas / Shapely.

It isn't immediately obvious from the [docs](https://tableau.github.io/hyper-db/docs/sql/scalar_func/geography), but it looks like you might need to run a SQL function on the geometry column to output it as WKB or WKT instead of whatever internal byte representation is used in HYPER.  Perhaps similar to something like [ST_AsBinary](https://postgis.net/docs/ST_AsBinary.html) in PostGIS.

**Reply by awa5114** [reply]
@brendan-ward  You were right, if I use the Hyper API SQL `CAST` function and cast the column to `text` type I get WKT strings which I was able to successfully convert to a geometry column.  So my comment about "ESRI proprietary format" was straight up nonsense. Thanks!
